netket/operator/_potential.py

```diff
@@ -43,14 +43,14 @@ def __init__(
 
         self._afun = afun
 
-        self.coefficient = jnp.array(coefficient, dtype=dtype)
+        self.coefficient = jnp.asarray(coefficient, dtype=dtype)
 
-        super().__init__(hilbert, self.coefficient.dtype)
+        super().__init__(hilbert, dtype)
 
     def _expect_kernel(
         self, logpsi: Callable, params: PyTree, x: Array, coefficient: Optional[PyTree]
     ):
-        return coefficient * self._afun(x)
+        return jnp.array(coefficient) * self._afun(x)
```

Comment:
you are converting in the initialisation to a jnp array so no need to do it again here
```suggestion
        return coefficient * self._afun(x)
```